### ISSUE TITLE
TOC not generated when using a particular Markdown header style

### ISSUE THREAD

**Issue by mrsmond** [doxygen/doxygen#6791]
Mainly as a note to others as this is not critical and has a work-around.

Doxygen v1.8.15

Table of contents (TOC) is not generated when using the === or --- style of headers in Markdown. It does generate correctly when using the # style of headers.

Example files attached. header_style_2.md (# style) works fine, however header_style_1.md:

```
Title {#mainpage}
===

[TOC]

One
===

Two
===

Three
===
```

Issues these warnings:

```
header_style_1.md:6: warning: multiple use of section label 'autotoc_md0' while adding section, (first occurrence: header_style_1.md, line 1)
header_style_1.md:9: warning: multiple use of section label 'autotoc_md1' while adding section, (first occurrence: header_style_1.md, line 1)
header_style_1.md:12: warning: multiple use of section label 'autotoc_md2' while adding section, (first occurrence: header_style_1.md, line 1)
```

As you can see from the Doxyfile, I have `TOC_INCLUDE_HEADINGS` > 0 as per #6513, which is related, but not the same problem. The extra header 1 section is needed due to bug #5892.

[toc_test.tar.gz](https://github.com/doxygen/doxygen/files/2779511/toc_test.tar.gz)

**Comment by mrsmond** [comment]
Interestingly, the [source](https://github.com/doxygen/doxygen/blob/master/doc/markdown.doc) for [the markdown documentation page](http://www.doxygen.nl/manual/markdown.html) has a [TOC] at the start, but there is no TOC in the HTML output.

**Comment by albert-github** [comment]
The problem regarding the 'multiple use of section label' this was indeed the case in v1.8.15 but has been solved in the pull request #6780  based on issue #6779 
The problem with `[TOC]` has to be investigated a bit further.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6792, regarding the `[TOC]`

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
